src/IotHub/IotHub/IotHub/Routing/RemoveAzureRmIotHubMessageEnrichment.cs

```diff
@@ -0,0 +1,117 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Management.IotHub
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.Management.IotHub.Common;
+    using Microsoft.Azure.Commands.Management.IotHub.Models;
+    using Microsoft.Azure.Management.IotHub;
+    using Microsoft.Azure.Management.IotHub.Models;
+    using ResourceManager.Common.ArgumentCompleters;
+
+    [Cmdlet("Remove", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "IotHubMessageEnrichment", DefaultParameterSetName = ResourceParameterSet, SupportsShouldProcess = true), OutputType(typeof(bool))]
+    [Alias("Remove-" + ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "IotHubMsgEnrich")]
+    public class RemoveAzureRmIotHubMessageEnrichment : IotHubBaseCmdlet
+    {
+        private const string ResourceIdParameterSet = "ResourceIdSet";
+        private const string ResourceParameterSet = "ResourceSet";
+        private const string InputObjectParameterSet = "InputObjectSet";
+
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = InputObjectParameterSet, ValueFromPipeline = true, HelpMessage = "IotHub object")]
+        [ValidateNotNullOrEmpty]
+        public PSIotHub InputObject { get; set; }
```

Comment:
The inputobject in this case should be PSEnrichmentMetadata.  This would allow the user to use piping scenarios like ```Get | Select | Remove```, which is a common pattern

Reply:
Model `EnrichmentProperties` of IotHub SDK doesn't expose the "Id" property. https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/iothub/Microsoft.Azure.Management.IotHub/src/Generated/Models/EnrichmentProperties.cs

Reply:
Then you should update the model to include it.  Allowing users to pipe Get | Remove is an important scenario.